pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/rule/DataClassRule.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.metrics.rule;
+
+import net.sourceforge.pmd.lang.java.ast.ASTAnyTypeDeclaration;
+import net.sourceforge.pmd.lang.java.metrics.JavaMetrics;
+import net.sourceforge.pmd.lang.java.metrics.api.JavaClassMetricKey;
+import net.sourceforge.pmd.lang.java.rule.AbstractJavaMetricsRule;
+
+/**
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+public class DataClassRule extends AbstractJavaMetricsRule {
+
+    // probably not worth using properties
+    private static final int ACCESSOR_OR_FIELD_FEW_LEVEL = 3;
+    private static final int ACCESSOR_OR_FIELD_MANY_LEVEL = 5;
+    private static final int WOC_LEVEL = 33;
+    private static final int WMC_HIGH_LEVEL = 31;
+    private static final int WMC_VERY_HIGH_LEVEL = 47;
+
+
+    @Override
+    public Object visit(ASTAnyTypeDeclaration node, Object data) {
+
+        boolean isDataClass = interfaceRevealsData(node) && classRevealsDataAndLacksComplexity(node);
+
+        if (isDataClass) {
+            addViolation(data, node, new String[] {node.getImage()});
+        }
+
+        return super.visit(node, data);
+    }
+
+
+    private boolean interfaceRevealsData(ASTAnyTypeDeclaration node) {
+        double woc = JavaMetrics.get(JavaClassMetricKey.WOC, node);
+        return woc < WOC_LEVEL;
```

Comment:
Yes you're right it's a proportion. Initially I multiplied the result by 100 in WocMetric (to get a percentage), and wanted to do the same for other proportion metrics. I changed my mind and decided it was better to output a normalised result, though I forgot to change this threshold back ^^